### ISSUE TITLE
Custom config on /etc/pve/nodes/[node]/lxc/[vm_id].conf

### ISSUE THREAD

**Issue by ersefa** [Telmate/terraform-provider-proxmox#533]
Hi
Thanks for your work in advance Telmate.
We have freeipa-client on our lxcs for ldap auth. This means we need to specify custom subuid / subgid addresses to manage ldap users on the lxc`s config files. Something like:

arch: amd64
cmode: tty
console: 1
cores: 4
cpulimit: 0
cpuunits: 1024
features: nesting=1
hostname: dataco
memory: 8192
net0: name=eth0,bridge=vmbr0,hwaddr=D2:C0:4E:AE:78:7C,ip=dhcp,ip6=dhcp,type=veth
onboot: 1
ostype: ubuntu
protection: 0
rootfs: local-zfs:subvol-119-disk-0,size=20G
swap: 0
tty: 2
unprivileged: 1
**lxc.idmap: u 100000 200000 100000
lxc.idmap: g 100000 200000 100000
lxc.idmap: u 0 100000 65536
lxc.idmap: g 0 100000 65536**

We are doing this "manually" with an ansible playbook but will be great if we can do this on the terraform step and have an operational machine just out of the box.
No clue if we can do this, nothing on doc related. But im sure that the provider looks at this config because when we add this with ansible, terraform complains that something has changed out of terraform.

Can give some advice? Thanks.

**Comment by mleone87** [comment]
Yes, changing .conf file reflects in a new data returned from the api		

```
"lxc": [
			[
				"lxc.idmap",
				"u 100000 200000 100000"
			],
			[
				"lxc.idmap",
				"g 100000 200000 100000"
			],
			[
				"lxc.idmap",
				"u 0 100000 65536"
			],
			[
				"lxc.idmap",
				"g 0 100000 65536"
			]
		],
```

I fear that requires a little bit of work from api client side and from provider side

**Comment by ersefa** [comment]
Then is not even implemented at promox api level right?

**Comment by mleone87** [comment]
actually is returned but cannot be set from api

**Comment by lhazlewood** [comment]
Hi there - is it possible to re-open this now that the API supports it? Using Proxmox 9.1:

`https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/lxc/{vmid}/config`

Since `pvesh` is just a perl HTTP client that wraps the API, this should be possible because you can do this from the command line (for example):

```bash
pvesh set /nodes/{nodename}/lxc/{vmid}/config --lxc.idmap='u 0 100000 65536'
```

Calling `GET` on the API shows the `lxc.idmap*` lines coming back as well.  For example:

```bash
$ pvesh get /nodes/$NODENAME/lxc/100/config
# ... stuff above here ...
├──────────────┼─────────────────────────────────────────────────────────────────────────────────
│ lxc          │ [["lxc.idmap","u 0 100000 65536"],["lxc.idmap","g 0 100000 65536"]]
├──────────────┼─────────────────────────────────────────────────────────────────────────────────
# ... stuff below here ...
```

Theoretically this _should_ be available on the `/nodes/$NODENAME/lxc` POST request as well so it can be done as a single request instead of requiring 1)`POST` on that endpoint and then 2) `PUT` (update) on the `/config` endpoint.
